typescript/looker/rtl/oauthSession.spec.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Looker Data Sciences, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+import { OAuthSession } from "./oauthSession";
+import {ApiSettings, DefaultSettings, IApiSettings} from "./apiSettings";
+import {BrowserCryptoHash, BrowserTransport} from "./browserTransport";
+import {NodeCryptoHash, NodeTransport} from "./nodeTransport";
+import {BrowserServices} from "./browserServices";
+import {MockCrypto} from "./mocks";
+import Url from "url";
+import {ITransport} from "./transport";
+
+describe('oauthSession', () => {
+  const settings = {
+    ...DefaultSettings(),
+    client_id: '123456',
+    redirect_uri: 'https://myapp.com/redirect',
+    base_url: 'https://looker.com:19999/api/3.1/',
+    looker_url: 'https://looker.com:9999',
```

Comment:
`base_url` is typically the value assigned to `looker_url`, and `apiPath` is constructed internally to be what you have here in `base_url` by plugging in API version into `base_url` concatenation, so I'm not sure `looker_url` is needed.

Reply:
Note the port numbers.  One is the url of the API web service, the other is the url of the Looker UI web service. I'd rather not assume one can be made from the other by merely changing port numbers, since port numbers can be changed by looker command line switches and by proxies.
The path is not needed for these tests; removed.